test/jdk/javax/swing/JComboBox/ComboPopupBug.java

```diff
@@ -21,66 +21,74 @@
-    private static final String instructionsText = """
-            This test is used to verify that clicking on JComboBox
-            when frame containing it is about to close should not
-            cause IllegalStateException.
-
-            A JComboBox is shown with Close button at the bottom.
-            Click on Close and then click on JComboBox arrow button
-            to try to show combobox popup.
-            If IllegalStateException is thrown, test will automatically Fail
-            otherwise click Pass.""";
+    private static JFrame frame;
+    private static JButton closeButton;
+    private static JComboBox<String> comboBox;
+    private static Point comboBoxLocation;
+    private static Dimension comboBoxSize;
+    private static Robot robot;
+    private static final int PADDING = 10;
 
     public static void main(String[] args) throws Exception {
-        PassFailJFrame.builder()
-                .title("ComboPopup Instructions")
-                .instructions(instructionsText)
-                .testTimeOut(5)
-                .rows(10)
-                .columns(35)
-                .testUI(ComboPopupBug::createUI)
-                .build()
-                .awaitAndCheck();
-    }
+        robot = new Robot();
+        SwingUtilities.invokeAndWait(() -> {
+            frame = new JFrame("ComboPopup");
+
+            comboBox = new JComboBox<>();
+            comboBox.setEditable(true);
+            comboBox.addItem("test");
+            comboBox.addItem("test2");
+            comboBox.addItem("test3");
 
-    private static JFrame createUI() {
-        JFrame frame = new JFrame("ComboPopup");
+            closeButton = new JButton("Close");
+            closeButton.addActionListener((e) -> {
+                        clickComboBox();
+                        frame.setVisible(false);
+                    });
 
-        JComboBox<String> cb = new JComboBox<>();
-        cb.setEditable(true);
-        cb.addItem("test");
-        cb.addItem("test2");
-        cb.addItem("test3");
+            frame.getContentPane().add(comboBox, "North");
+            frame.getContentPane().add(closeButton, "South");
+            frame.setSize(200, 200);
+            frame.setVisible(true);
+        });
 
-        JButton b = new JButton("Close");
-        b.addActionListener(
-                (e)->{
-                    try {
-                        Thread.sleep(3000);
-                    } catch (Exception ignored) {
-                    }
-                    frame.setVisible(false);
-                });
+        robot.waitForIdle();
+        robot.delay(1000);
+
+        try {
+            SwingUtilities.invokeAndWait(() -> closeButton.doClick());
+        }
+        finally {
+            SwingUtilities.invokeAndWait(() -> frame.dispose());
+        }
+    }
 
-        frame.getContentPane().add(cb, "North");
-        frame.getContentPane().add(b, "South");
-        frame.setSize(200, 200);
+    public static void clickComboBox() {
```

Comment:
```suggestion
    private static void clickComboBox() {
```
Not that important, yet I prefer having all the methods `private`.